openvpn3/session.go

```diff
@@ -100,35 +102,44 @@ var ErrConnectFailed = errors.New("openvpn3 connect failed")
 
 // Start starts the session
 func (session *Session) Start() {
+	session.mutex.Lock()
```

Comment:
This approach is not what I actually expected.
My idea was to reduce runtime variables (finished, started, sessionPtr), not to add more.
The problem with stop was that C function is called directly instead of some kind of signaling mechanism.
Maybe a better approach would be:
Start() function starts go routine with inits, etc. Go routine blocks since C.start_session is blocking and returns exit error
another go routine is started which waits for stop signal( waitgroup would be perfect here) and when signal arrives - calls C.stop_session. Since we start this routine after we obtain session pointer - no need to worry that it might be null anymore.
Stop function simply signals that stop is requested - stop goroutine calls C.stop_session and exits. No more stops are possible to do on the same C pointer

Bonus points:
I guess only one runtime var is left - possible error returned from C session. All other vars can be init at construction time - so safe to use. Stop functions is simplified (no mutexes, no isStarted ugly checks). since session pointer can be function local variable - it can be removed from structure